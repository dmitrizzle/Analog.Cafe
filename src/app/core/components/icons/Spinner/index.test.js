import { shallow } from "enzyme"
import React from "react"

import Spinner from "./"

it("Render Spinner without crashing", () => {
  shallow(<Spinner />)
})
