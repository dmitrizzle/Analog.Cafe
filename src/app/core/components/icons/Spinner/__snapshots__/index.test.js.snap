// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Spinner, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <svg
        viewBox="28 22 20 20"
        xmlns="http://www.w3.org/2000/svg"
>
        <defs>
                <clipPath
                        id="loader-clip"
                >
                        <rect
                                height="40"
                                width="44"
                                x="20"
                                y="0"
                        />
                </clipPath>
        </defs>
        <path
                clipPath="url(#loader-clip)"
                d="M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3"
                fill="none"
                stroke="#fff"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="2"
        >
                <animateTransform
                        attributeName="transform"
                        begin="0s"
                        calcMode="linear"
                        dur="750ms"
                        keyTimes="0;1"
                        repeatCount="indefinite"
                        type="translate"
                        values="-20 0;7 0"
                />
                <animate
                        attributeName="stroke-dasharray"
                        begin="0s"
                        calcMode="linear"
                        dur="750ms"
                        keyTimes="0;1"
                        repeatCount="indefinite"
                        values="0 20 23 60;0 69 22.5 10"
                />
        </path>
</svg>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <defs>
            <clipPath
                        id="loader-clip"
            >
                        <rect
                                    height="40"
                                    width="44"
                                    x="20"
                                    y="0"
                        />
            </clipPath>
</defs>,
          <path
            clipPath="url(#loader-clip)"
            d="M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3"
            fill="none"
            stroke="#fff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
>
            <animateTransform
                        attributeName="transform"
                        begin="0s"
                        calcMode="linear"
                        dur="750ms"
                        keyTimes="0;1"
                        repeatCount="indefinite"
                        type="translate"
                        values="-20 0;7 0"
            />
            <animate
                        attributeName="stroke-dasharray"
                        begin="0s"
                        calcMode="linear"
                        dur="750ms"
                        keyTimes="0;1"
                        repeatCount="indefinite"
                        values="0 20 23 60;0 69 22.5 10"
            />
</path>,
        ],
        "viewBox": "28 22 20 20",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <clipPath
              id="loader-clip"
>
              <rect
                            height="40"
                            width="44"
                            x="20"
                            y="0"
              />
</clipPath>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <rect
                height="40"
                width="44"
                x="20"
                y="0"
/>,
              "id": "loader-clip",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "height": "40",
                "width": "44",
                "x": "20",
                "y": "0",
              },
              "ref": null,
              "rendered": null,
              "type": "rect",
            },
            "type": "clipPath",
          },
          "type": "defs",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <animateTransform
                attributeName="transform"
                begin="0s"
                calcMode="linear"
                dur="750ms"
                keyTimes="0;1"
                repeatCount="indefinite"
                type="translate"
                values="-20 0;7 0"
/>,
              <animate
                attributeName="stroke-dasharray"
                begin="0s"
                calcMode="linear"
                dur="750ms"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0 20 23 60;0 69 22.5 10"
/>,
            ],
            "clipPath": "url(#loader-clip)",
            "d": "M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3",
            "fill": "none",
            "stroke": "#fff",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeMiterlimit": "10",
            "strokeWidth": "2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "attributeName": "transform",
                "begin": "0s",
                "calcMode": "linear",
                "dur": "750ms",
                "keyTimes": "0;1",
                "repeatCount": "indefinite",
                "type": "translate",
                "values": "-20 0;7 0",
              },
              "ref": null,
              "rendered": null,
              "type": "animateTransform",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "attributeName": "stroke-dasharray",
                "begin": "0s",
                "calcMode": "linear",
                "dur": "750ms",
                "keyTimes": "0;1",
                "repeatCount": "indefinite",
                "values": "0 20 23 60;0 69 22.5 10",
              },
              "ref": null,
              "rendered": null,
              "type": "animate",
            },
          ],
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <svg
          viewBox="28 22 20 20"
          xmlns="http://www.w3.org/2000/svg"
>
          <defs>
                    <clipPath
                              id="loader-clip"
                    >
                              <rect
                                        height="40"
                                        width="44"
                                        x="20"
                                        y="0"
                              />
                    </clipPath>
          </defs>
          <path
                    clipPath="url(#loader-clip)"
                    d="M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3"
                    fill="none"
                    stroke="#fff"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit="10"
                    strokeWidth="2"
          >
                    <animateTransform
                              attributeName="transform"
                              begin="0s"
                              calcMode="linear"
                              dur="750ms"
                              keyTimes="0;1"
                              repeatCount="indefinite"
                              type="translate"
                              values="-20 0;7 0"
                    />
                    <animate
                              attributeName="stroke-dasharray"
                              begin="0s"
                              calcMode="linear"
                              dur="750ms"
                              keyTimes="0;1"
                              repeatCount="indefinite"
                              values="0 20 23 60;0 69 22.5 10"
                    />
          </path>
</svg>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <defs>
              <clipPath
                            id="loader-clip"
              >
                            <rect
                                          height="40"
                                          width="44"
                                          x="20"
                                          y="0"
                            />
              </clipPath>
</defs>,
            <path
              clipPath="url(#loader-clip)"
              d="M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3"
              fill="none"
              stroke="#fff"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="2"
>
              <animateTransform
                            attributeName="transform"
                            begin="0s"
                            calcMode="linear"
                            dur="750ms"
                            keyTimes="0;1"
                            repeatCount="indefinite"
                            type="translate"
                            values="-20 0;7 0"
              />
              <animate
                            attributeName="stroke-dasharray"
                            begin="0s"
                            calcMode="linear"
                            dur="750ms"
                            keyTimes="0;1"
                            repeatCount="indefinite"
                            values="0 20 23 60;0 69 22.5 10"
              />
</path>,
          ],
          "viewBox": "28 22 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <clipPath
                id="loader-clip"
>
                <rect
                                height="40"
                                width="44"
                                x="20"
                                y="0"
                />
</clipPath>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <rect
                  height="40"
                  width="44"
                  x="20"
                  y="0"
/>,
                "id": "loader-clip",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "height": "40",
                  "width": "44",
                  "x": "20",
                  "y": "0",
                },
                "ref": null,
                "rendered": null,
                "type": "rect",
              },
              "type": "clipPath",
            },
            "type": "defs",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <animateTransform
                  attributeName="transform"
                  begin="0s"
                  calcMode="linear"
                  dur="750ms"
                  keyTimes="0;1"
                  repeatCount="indefinite"
                  type="translate"
                  values="-20 0;7 0"
/>,
                <animate
                  attributeName="stroke-dasharray"
                  begin="0s"
                  calcMode="linear"
                  dur="750ms"
                  keyTimes="0;1"
                  repeatCount="indefinite"
                  values="0 20 23 60;0 69 22.5 10"
/>,
              ],
              "clipPath": "url(#loader-clip)",
              "d": "M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3",
              "fill": "none",
              "stroke": "#fff",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeMiterlimit": "10",
              "strokeWidth": "2",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "attributeName": "transform",
                  "begin": "0s",
                  "calcMode": "linear",
                  "dur": "750ms",
                  "keyTimes": "0;1",
                  "repeatCount": "indefinite",
                  "type": "translate",
                  "values": "-20 0;7 0",
                },
                "ref": null,
                "rendered": null,
                "type": "animateTransform",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "attributeName": "stroke-dasharray",
                  "begin": "0s",
                  "calcMode": "linear",
                  "dur": "750ms",
                  "keyTimes": "0;1",
                  "repeatCount": "indefinite",
                  "values": "0 20 23 60;0 69 22.5 10",
                },
                "ref": null,
                "rendered": null,
                "type": "animate",
              },
            ],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
